### PR TITLE
Apply access control and nonce verification

### PR DESCRIPTION
**Pull request by stephywells**
Fixes https://github.com/Strategy11/awpcp/issues/3158

It applies access control with current_user_can() and nonce verification with wp_verify_nonce() to the vulnerable function. The changes include adding checks on ajax requests, checking nonce on field order, and more ajax permissions checked. This pull request also includes PHP 8 deprecation fixes and updates. 

Things that could be affected that need testing:
- ajax functions related to credit plans table, setting credits on WP users table, quick start dismissal, SSL check on debug page, import, export, dismiss admin notices.
- creating a plan, selecting a category in the new listing form, selecting a plan in the new listing form
- Verify that creating and editing a listing works on the front-end

## Summary by CodeRabbit

- **New Features**
	- Improved PHP 8 support and enhanced security measures in the AWP Classifieds plugin.
	- Added dynamic script loading with support for minified versions based on debug status.
	- Introduced nonce and user role checks for admin AJAX actions.
- **Bug Fixes**
	- Fixed currency symbol formatting in test functions to ensure proper display.
	- Addressed conflicts with Select2 by falling back to native dropdowns to prevent CSRF attacks in admin pages.
- **Refactor**
	- Simplified initialization logic for current transactions in the payments API.
	- Minor adjustments to padding in IE-6 specific CSS for better alignment.
- **Documentation**
	- Updated README.TXT to reflect new features and improvements.
- **Chores**
	- Updated plugin version to 4.3.2, including metadata and database version alignment.
- **Tests**
	- Added a new test file requirement in the bootstrap setup for enhanced testing coverage.

### CODE REVIEW

File: functions.php [hunk 1]
```diff
@@ -520,6 +520,18 @@ function awpcp_user_is_admin($id) {
...
+function awpcp_check_admin_ajax() {
+    check_ajax_referer( 'awpcp_ajax', 'nonce' );
+    if ( ! awpcp_current_user_is_admin() ) {
+        wp_die( esc_html__( 'You are not authorized to perform this action.', 'another-wordpress-classifieds-plugin' ) );
+    }
+}
+
 function awpcp_get_grid_item_css_class($classes, $pos, $columns, $rows) {
 	if ($pos < $columns)
 		$classes[] = 'first-row';
```
[the remark this answers is not in the document]
Comment (stephywells) [reply functions.php:537]: This is unrelated to the pr.

File: includes/class-awpcp.php [hunk 2]
```diff
@@ -1109,7 +1111,7 @@ public function register_scripts() {
         // https://plugins.svn.wordpress.org/buddypress/tags/3.1.0/bp-core/bp-core-cssjs.php
         wp_register_script(
             'awpcp-moment-with-locales',
-            "{$vendors}/moment-2.22.2/moment-with-locales.min.js",
+            $vendors . '/moment-2.22.2/moment-with-locales' . $min . '.js',
```
[the remark this answers is not in the document]
Comment (stephywells) [reply includes/class-awpcp.php:1114]: Done

**Comment by garretlaxton** [comment]
When I go to add and edit a listing, the form is missing for me.
image
image

I can see the form in the page source, but nothing is there.
image

File: frontend/class-awpcp-meta.php [hunk 3]
```diff
@@ -23,6 +23,7 @@ class AWPCP_Meta {
     private $request = null;
 
     private $doing_opengraph = false;
+    public $doin_description_meta_tag = false;
```
[the remark this answers is not in the document]
Comment (stephywells) [reply frontend/class-awpcp-meta.php:26]: That would be nice, but this is legacy code.

**Comment by stephywells** [comment]
Oh! Do you have SCRIPT_DEBUG on? It's possible you're missing the js files this is using now. I'll check that out.

**Comment by garretlaxton** [comment]
I did not have it enabled. Let me know if or when this is ready again.

**Comment by stephywells** [comment]
@garretlaxton Can you please give me more detail on how I might replicate the issue you are seeing?

**Comment by garretlaxton** [comment]
I notice this error in the console. I also enabled script_debug.
image

I also tried it on a new fresh site with the same results. I also switched between PHP 7.4 and 8.0, but no difference.

Here is my tests site if you want to look there.
https://grunge.us.eu.org/classifieds/place-ad/
https://grunge.us.eu.org/wp-admin/
admin
cQt@2n^37kzYD@WsL

**Comment by stephywells** [comment]
Thanks @garretlaxton ! Found the issue with Script Debug off.

I removed the password from your comment since this is public, but you may want to change the password too.

**Pull request merged**